ui/src/pageLayout/containers/TreeNav.tsx

```diff
@@ -67,132 +74,134 @@ class TreeSidebar extends PureComponent<Props> {
     const navItems = generateNavItems(orgID)
 
     return (
-      <TreeNav
-        expanded={isExpanded}
-        headerElement={<NavHeader link={orgPrefix} />}
-        userElement={<UserWidget />}
-        onToggleClick={handleToggleNavExpansion}
-        bannerElement={<CloudUpgradeNavBanner />}
-      >
-        {navItems.map(item => {
-          const linkElement = (className: string): JSX.Element => {
-            if (item.link.type === 'href') {
-              return <a href={item.link.location} className={className} />
+      <OrgSettings>
+        <TreeNav
+          expanded={isExpanded}
+          headerElement={<NavHeader link={orgPrefix} />}
+          userElement={<UserWidget />}
+          onToggleClick={handleToggleNavExpansion}
+          bannerElement={showUpgradeButton ? <CloudUpgradeNavBanner /> : null}
```

Comment:
why add the concept of "showUpgradeButton" here, and not in the `CloudUpgradeNavBanner` component? from a quick read, it seems like it would unnecessarily couple the two components when `CloudUpgradeNavButton` could just return null

Reply:
Originally, I had the show logic inside the ```CloudUpgradeNavBanner``` but I had a tough time getting it to work because of the React Fragment in there. I will make another attempt to move it back in there.

Reply:
Ok I got it working this time.

`TreeSideBar` is still responsible for fetching the settings. Since we can reasonably expect that there will never be more than one `TreeSideBar`, we can also reasonably expect that we will never make unnecessary fetches for the same resource.

`CloudUpgradeButton` and `CloudUpgradeNavBanner` can hide themselves based on the state props passed in.